tests/test_iso_tx_basic/src/main.xc

```diff
@@ -1,61 +1,54 @@
 // Copyright 2016-2021 XMOS LIMITED.
 // This Software is subject to the terms of the XMOS Public Licence: Version 1.
-/*
- * Test the use of the ExampleTestbench. Test that the value 0 and 1 can be sent
- * in both directions between the ports.
- *
- * NOTE: The src/testbenches/ExampleTestbench must have been compiled for this to run without error.
- *
- */
-#include <xs1.h>
-#include <print.h>
-#include <stdio.h>
-#include "xud.h"
-#include "platform.h"
+
 #include "shared.h"
 
-#ifndef XUD_EP_COUNT_OUT
-#define XUD_EP_COUNT_OUT   (4)
+#ifndef EP_COUNT_OUT
+#define EP_COUNT_OUT   (6)
 #endif
 
-#ifndef XUD_EP_COUNT_IN
-#define XUD_EP_COUNT_IN    (4)
+#ifndef EP_COUNT_IN
+#define EP_COUNT_IN    (6)
 #endif
 
 /* Endpoint type tables */
-XUD_EpType epTypeTableOut[XUD_EP_COUNT_OUT] = {XUD_EPTYPE_CTL,
+XUD_EpType epTypeTableOut[EP_COUNT_OUT] = {XUD_EPTYPE_CTL,
                                                 XUD_EPTYPE_BUL,
+                                                 XUD_EPTYPE_BUL,
+                                                 XUD_EPTYPE_BUL,
                                                  XUD_EPTYPE_BUL,
                                                  XUD_EPTYPE_BUL};
-XUD_EpType epTypeTableIn[XUD_EP_COUNT_IN] =   {XUD_EPTYPE_CTL, 
-                                                XUD_EPTYPE_BUL,
-                                                XUD_EPTYPE_BUL,
+XUD_EpType epTypeTableIn[EP_COUNT_IN] =   {XUD_EPTYPE_CTL, 
+                                                XUD_EPTYPE_ISO,
+                                                XUD_EPTYPE_ISO,
+                                                XUD_EPTYPE_ISO,
+                                                XUD_EPTYPE_ISO,
                                                 XUD_EPTYPE_ISO};
 
 int main()
 {
-    chan c_ep_out[XUD_EP_COUNT_OUT], c_ep_in[XUD_EP_COUNT_IN];
-    chan c_dummy[DUMMY_THREAD_COUNT];
+    chan c_ep_out[EP_COUNT_OUT], c_ep_in[EP_COUNT_IN];
+    //chan c_dummy[DUMMY_THREAD_COUNT];
```

Comment:
Any reason to keep the commented-out stuff?

Reply:
This needs re-instating (as per issue #114). However, the current scheme fails due to running out of channel ends. Need to come up with a different scheme.